app/controllers/products_controller.rb

```diff
@@ -0,0 +1,44 @@
+class ProductsController < ApplicationController
+
+  def new
+    @product = Product.new
+  end
+
+  def edit
+    id = params[:id]
+    @product = Product.find(id)
+    @title = "Edit Product"
+    @action = :update
+  end
+
+  def update
+    id = params[:id]
+    product = Product.find(id)
+    vendor_id = product.vendor_id
+    Product.update(params[:id], product_params[:product])
```

Comment:
You have a local variable for the `id` that should be used instead of `params[:id]`